packages/core/src/payment/strategies/paypal-commerce/paypal-commerce-script-loader.ts

```diff
@@ -8,20 +8,29 @@ import { FundingType, PaypalCommerceHostWindow, PaypalCommerceInitializationData
 
 export default class PaypalCommerceScriptLoader {
     private _window: PaypalCommerceHostWindow;
+    private _paypalSdk?: Promise<PaypalCommerceSDK>;
 
     constructor(
         private _scriptLoader: ScriptLoader
     ) {
         this._window = window;
     }
 
-    async loadPaypalCommerce(
+    async getPayPalSDK(
         paymentMethod: PaymentMethod<PaypalCommerceInitializationData>,
         currencyCode: string,
         initializesOnCheckoutPage?: boolean,
     ): Promise<PaypalCommerceSDK> {
-        const paypalSdkScriptConfig = this._getPayPalSdkScriptConfigOrThrow(paymentMethod, currencyCode, initializesOnCheckoutPage);
+        if (!this._paypalSdk) {
```

Comment:
Just out of curiosity, is this an effort to cache the paypal sdk since it's being used by multiple clients?

Reply:
Yes. In this case the method will return the same Promise for all initialized strategies what use PayPal SDK loader